cybershell/memory.py

```
@@ -0,0 +1,74 @@
+    def transform(self, docs: List[str]):
+        vecs = []
+        for doc in docs:
+            counts = defaultdict(int)
+            for t in self._tok(doc): counts[t]+=1
+            vec = [0.0]*len(self.vocab)
+            for t,c in counts.items():
+                if t in self.vocab: vec[self.vocab[t]] = float(c)
+            vecs.append(vec)
+        return vecs
+    @staticmethod
+    def cosine(a,b):
```

Ruff (0.12.2)

Missing return type annotation for staticmethod `cosine`

(ANN205)